File: leetcode/java/0232. Implement Queue using Stacks.java
package leetcode.java;

import java.util.Stack;

class MyQueue {

    public Stack<Integer> s1;
    public Stack<Integer> s2;

    public MyQueue() {
        this.s1 = new Stack<>();
        this.s2 = new Stack<>();
    }

    public void push(int x) {
        this.s1.push(x);
    }
    
    public int pop() {
        if (this.s2.size() == 0) {
            while (this.s1.size() > 1) {
                this.s2.add(this.s1.pop());
            }
            return this.s1.pop();
        } else {
            return this.s2.pop();
        }
    }
    
    public int peek() {
        if (this.s2.size() == 0) {
            while (this.s1.size() > 0) {
                this.s2.add(this.s1.pop());
            }
            return this.s1.peek();
        }
        return this.s2.peek();
    }
    
    public boolean empty() {
        return this.s1.size() == 0 && this.s2.size() == 0;
    }
}

File: leetcode/python/0232. Implement Queue using Stacks.py
'''
First stack is the default intake stack
Second stack is used to reverse the order. All pops will be made from this stack
'''

class MyQueue(object):

    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def push(self, x):
        self.stack1.append(x)
        

    def pop(self):
        if len(self.stack2) > 0:
            return self.stack2.pop()
        else:
            while len(self.stack1) > 0:
                self.stack2.append(self.stack1.pop())
            return self.stack2.pop()

    def peek(self):
        if len(self.stack2) > 0:
            return self.stack2[-1]
        else:
            while len(self.stack1) > 0:
                self.stack2.append(self.stack1.pop())
            return self.stack2[-1]
        

    def empty(self):
        return len(self.stack1) == 0 and len(self.stack2) == 0